bash_completion.sh

```diff
@@ -218,15 +243,15 @@ _buildtest ()
       COMPREPLY=( $( compgen -W "${cmds}" -- $cur ) )
       # handle completion logic for 'buildtest config <subcommand>' based on subcommands
 
-      case "${COMP_WORDS[COMP_CWORD-2]}" in
+      case "${COMP_WORDS[offset_by_two]}" in
```

Comment:
looking at the commit diff the `COMP_CWORDS` does get the total words and thats what we set for current pointer and previous.

```bash
local cur="${COMP_WORDS[COMP_CWORD]}"
local prev="${COMP_WORDS[COMP_CWORD-1]}"
```  

I found this https://devmanual.gentoo.org/tasks-reference/completion/index.html that explains about the different variables we can use including COMP_LINE, COMP_POINT. I think that we need to get this logic correct and inspect the variable COMP_WORDS to make sure we index the correct tab completion. 

I apologize for my last suggestion, the variable `offset` makes sense now that i think about it but having `offset_by_two` and `offset_by_three` doesnt make sense. Can you review this logic and try again.

Reply:
@shahzebsiddiqui I had tried earlier using `COMP_CWORD` but getting the index based on the total words will not work in every case. The index would be incorrect if there is another subcommad/option ahead. For instance, doing `COMP_WORDS[COMP_CWORD-1]` would makes sense for `buildtest cg co` it gives the index of `co` option but if we now do `buildtest cg co find` then `COMP_WORDS[COMP_CWORD-1]` would give the index of `find` option instead of the index of `co`. So the code will never enter into any of the parent case statements for previous commands.

Reply:
i understand. I think for now this is a slight improvement from the current setup since we couldn't get bash completion working when top-level options were specified. That being said your latest change which reverted the changes back look better so i am fine with merging this in.

Thanks